### PR TITLE
Add SoupX

### PR DESCRIPTION
**Pull request by mvfki**
There is still an R CMD check error from BCDS on my local. Not sure whether there's some environment configuration issue on my side.

### CODE REVIEW

File: R/runSoupX.R [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        }
+    } else {
+        message(paste0(date(), " ... Cluster info not given, "))
+        message(paste0(date(), " ...   generating with Scran SNN method."))
+        suppressMessages({
+            inSCE <- scaterlogNormCounts(inSCE, "logcounts", useAssay)
+            inSCE <- runNormalization(inSCE, "logcounts", "scaled", scale = TRUE)
+            inSCE <- seuratFindHVG(inSCE, verbose = FALSE)
+            inSCE <- getTopHVG(inSCE, "vst", altExp = "hvg")
+            inSCE <- scaterPCA(inSCE, "scaled", "hvg")
```
Comment (joshua-d-campbell) [inline R/runSoupX.R:258]: what about using the "quickCluster" from scater? Not sure what works better. Does soupx have a recommendation? @mvfki

File: R/runSoupX.R [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    
+    # Adding cluster info
+    if (!is.null(cluster)) {
+        if (is.character(cluster) & length(cluster) == 1) {
+            cluster <- SummarizedExperiment::colData(inSCE)[[cluster]]
+            names(cluster) <- colnames(inSCE)
+            sc = SoupX::setClusters(sc, cluster)
+        } else if (length(cluster) == ncol(inSCE)) {
+            names(cluster) <- colnames(inSCE)
+            sc = SoupX::setClusters(sc, cluster)
```
[the remark this answers is not in the document]
Comment (mvfki) [reply R/runSoupX.R:246]: Will include this in next push

File: R/runSoupX.R (hunk 1 above)
Comment (mvfki) [reply R/runSoupX.R:258]: I'll definitely try this out. 
They haven't recommended any method but just saying "The results are not strongly sensitive to the clustering used". But in my own test, clustering in the way I implemented works better than running `runScranSNN(..., useAssay = "counts")`

Comment (mvfki) [reply R/runSoupX.R:258]: > what about using the "quickCluster" from scater? Not sure what works better. Does soupx have a recommendation? @mvfki

By the way, I just googled it, did you mean "quickCluster" from Scran?

Comment (joshua-d-campbell) [reply R/runSoupX.R:258]: yea!

**Pull request merged**